legion/engine/physics/physx/physx_integration_helpers.cpp

```diff
@@ -0,0 +1,85 @@
+#include "physx_integration_helpers.hpp"
+#include <physics/physx/systems/physx_physics_system.hpp>
+#include <physx/PxPhysicsAPI.h>
+
+using namespace physx;
+
+namespace legion::physics
+{
+    physx::PxPhysics* getSDK()
+    {
+        return PhysXPhysicsSystem::getSDK();
+    }
+
+    static inline void toPhysxTransform(physx::PxTransform& pxTransform, const math::vec3& pos, const math::quat& rot)
+    {
+        pxTransform.p = { pos.x,pos.y,pos.z };
+        pxTransform.q = { rot.x,rot.y,rot.z,rot.w };
+    }
+
+    void calculateGlobalAndLocalTransforms(physx::PxTransform& outLocalTransform, physx::PxTransform& outGlobalTransform, const ColliderData& collider, ecs::entity ent)
+    {
+        const math::vec3& pos = *ent.get_component<position>();
+        const math::quat& rot = *ent.get_component<rotation>();
+
+        const math::vec3& localOffset = collider.getOffset();
+        const math::quat& localRot = collider.getRotationOffset();
+
+        toPhysxTransform(outGlobalTransform, pos, rot);
+        toPhysxTransform(outLocalTransform, localOffset, localRot);
+    }
+
+    template<class PxGeometry, class ...GeometryArgs>
+    void instantiateStaticActorWith(physx::PxPhysics* sdk, PhysxInternalWrapper& wrapper,
+        const physx::PxTransform& globalTransform, const physx::PxTransform& localTransform, const PhysxEnviromentInfo& sceneInfo, ecs::entity ent, GeometryArgs... geometryArgs)
+    {
+        PxRigidStatic* staticCollider = PxCreateStatic(*getSDK(), globalTransform, 
+            PxGeometry(std::forward<GeometryArgs>(geometryArgs)...), *sceneInfo.defaultMaterial, localTransform);
+        staticCollider->userData = ent.data;
+
+        wrapper.physicsActor = staticCollider;
+        sceneInfo.scene->addActor(*staticCollider);
+    }
+
+    template<class PxGeometry, class ...GeometryArgs>
+    void instantiateDynamicActorWith(physx::PxPhysics* sdk, PhysxInternalWrapper& wrapper,
+        const physx::PxTransform& globalTransform, const physx::PxTransform& localTransform, const PhysxEnviromentInfo& sceneInfo, ecs::entity ent, GeometryArgs... geometryArgs)
+    {
+        PxRigidDynamic* rb = PxCreateDynamic(*getSDK(), globalTransform,
+            PxGeometry(std::forward<GeometryArgs>(geometryArgs)...), *sceneInfo.defaultMaterial, 1.0f, localTransform);
+        rb->userData = ent.data;
+
+        wrapper.physicsActor = rb;
+        sceneInfo.scene->addActor(*rb);
+    }
+
+
+    template<class PxGeometry, class ...GeometryArgs>
+    void instantiateNextCollider(physx::PxPhysics* sdk, PhysxInternalWrapper& wrapper, const physx::PxTransform& globalTransform, const physx::PxTransform& localTransform,
+        const PhysxEnviromentInfo& sceneInfo, GeometryArgs ...geometryArgs)
+    {
+        PxShape* shape = getSDK()->createShape(PxGeometry(std::forward<GeometryArgs>(geometryArgs)...), *sceneInfo.defaultMaterial, true);
+
+        PxRigidActor* rigid = static_cast<PxRigidActor*>(wrapper.physicsActor);
```

Comment:
do we not need to check whether this is a rigidbody actor first?

Reply:
static colliders (PxRigidStatic) and rigidbody actors (PxRigidDynamic) are derived from PxRigidActor so a check isn't necessary